### ISSUE TITLE
[Bug] Layout does not react to tap with CollectionView child on Android

### ISSUE THREAD

**Issue by vikern22** [xamarin/Xamarin.Forms#9987]
# Description

The layout containing a CollectionView does not react to taps in the area where the CollectionView is present on the Android platform. Seems to work fine on the iOS platform.

### Steps to Reproduce

1. Run the attached sample on Android
2. Tap anywhere on the CollectionView

### Expected Behavior

The tapped event is fired and the Label below should display "Tapped!"

### Actual Behavior

The tapped event is not fired.

### Basic Information

- Version with issue: 4.5.0.396
- Last known good version:
- IDE: VS 2019 Enterprise
- Platform Target Frameworks: 
  - iOS:  13.3
  - Android: 10
- Nuget Packages: NETStandard.Library, Xamarin.Essentials, Xamarin.Forms
- Affected Devices: Pixel 2 emulator, Galaxy Note 4

### Reproduction Link

[https://drive.google.com/open?id=1Ci6NjRY9KvE3RbYSDZQ4f0pV1IsGY_LN](url)

**Comment by jfversluis** [comment]
Adding the reproduction to the GitHub issue so it won't go lost on us.

If I understand your problem correctly, you have a `StackLayout` with a `TapGestureRecognizer` and that recognizer does not fire when you add a `CollectionView` inside that `StackLayout`, right?

That is what I see happening. This works for iOS, but not Android. When I change the `StackLayout` to a `Grid` it does seem to work.

[CollectionViewTest.zip](https://github.com/xamarin/Xamarin.Forms/files/4343993/CollectionViewTest.zip)

**Comment by vikern22** [comment]
Correct. Does not seem to work if I programatically add a TapGestureRecognizer to the CollectionView either.

**Comment by vikern22** [comment]
Does not work replacing the StackLayout with a Grid for me.
